addons/maintainence-agent/maintenance_mode_controller.go

```diff
@@ -122,62 +130,88 @@ func (r *MaintenanceModeReconciler) startMaintenanceActions(ctx context.Context,
 				klog.Info("Scaling up RBD mirror deployments")
 				replicas = 1
 			}
-			_, err := r.scaleRBDMirrorDeployment(ctx, clusters, replicas)
+			_, err := r.scaleRBDMirrorDeployment(ctx, *cluster, replicas)
```

Comment:
Ignoring returned result can be an issue, if that requests a requeue and no error, for example potentially from line 154

Reply:
Changed to return the requeue result back .